client/src/controllers/SyncController.ts

```diff
@@ -0,0 +1,129 @@
+ *     id="event-date"
+ *     name="event-date"
+ *     value="2025-07-22"
+ *     data-controller="w-sync"
+ *     data-action="change->w-sync#apply cut->w-sync#clear"
+ *     data-w-sync-target-value="#title"
+ *   />
+ * </section>
+ */
+export class SyncController extends Controller<HTMLInputElement> {
+  static values = {
+    delay: { default: 0, type: Number },
+    disabled: { default: false, type: Boolean },
+    quiet: { default: false, type: Boolean },
+    target: String,
+  };
+
+  declare delayValue: number;
+  declare disabledValue: boolean;
+  declare quietValue: boolean;
+  declare readonly targetValue: string;
+
+  /**
+   * Dispatches an event to all target elements so that they can be notified
+   * that a sync has started, allowing them to disable the sync by preventing
+   * default.
+   */
+  connect() {
+    this.processTargetElements('start', true);
+  }
+
+  /**
+   * Allows for targeted elements to determine, via preventing the default event,
+   * whether this sync controller should be disabled.
+   */
+  check() {
+    this.processTargetElements('check', true);
+  }
+
+  /**
+   * Applies a value from the controlled element to the targeted
+   * elements.
+   */
+  apply() {
+    this.processTargetElements('apply').forEach((target) => {
+      setTimeout(() => {
+        target.setAttribute('value', this.element.value);
+
+        if (this.quietValue) return;
+        this.dispatch('change', {
+          cancelable: false,
+          prefix: '',
+          target: target as HTMLInputElement,
+        });
+      }, this.delayValue);
+    });
+  }
+
+  /**
+   * Clears the value of the targeted elements.
+   */
+  clear() {
+    this.processTargetElements('clear').forEach((target) => {
+      setTimeout(() => {
+        target.setAttribute('value', '');
+        if (this.quietValue) return;
+        this.dispatch('change', {
+          cancelable: false,
+          prefix: '',
+          target: target as HTMLInputElement,
+        });
+      }, this.delayValue);
+    });
+  }
+
+  /**
+   * Simple method to dispatch a ping event to the targeted elements.
+   */
+  ping() {
+    this.processTargetElements('ping', false, { bubbles: true });
+  }
+
+  /**
+   * Returns the non-default prevented elements that are targets of this sync
+   * controller.
+   */
+  processTargetElements(
+    eventName: string,
+    resetDisabledValue = false,
+    options = {},
+  ) {
+    if (!resetDisabledValue && this.disabledValue) {
+      return [];
+    }
+
+    const targetElements = [
+      ...document.querySelectorAll<HTMLElement>(this.targetValue),
+    ];
+
+    const elements = targetElements.filter((target) => {
+      const event = this.dispatch(eventName, {
+        bubbles: false,
+        cancelable: true,
+        ...options, // allow overriding some options but not detail & target
+        detail: { element: this.element, value: this.element.value },
+        target: target as HTMLInputElement, // BUG REF
```

Comment:
I was actually skeptical about this
Because when I checked the Stimulus doc, setting the target key as `target` and `value` as the target html was how they did it in their docs

Reply:
Yeah. This is not ideal but otherwise TypeScript thinks you are passing in the wrong element. When in fact it should allow any element/Node.